aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -218,6 +223,11 @@ impl VMChangeSet {
     }
 
     pub(crate) fn into_storage_change_set_unchecked(self) -> StorageChangeSet {
+        // It is still invalid to convert VMChangeSet that was created in a mode
+        // that produces DelayedField or ResourceGroup write sets
+        assert!(self.delayed_field_change_set().is_empty());
+        assert!(self.resource_group_write_set().is_empty());
```

Comment:
Can you please expand the comment a little more on why it is invalid to convert VMChangeSet when the resource group write set is non-empty.